### DISCUSSION TITLE
Dynamic tooltip text?

### DISCUSSION THREAD

**Discussion by nijlandvpr** [alpinejs/alpine#2047] [1. Help]
I'm developing with Alpine and Livewire and discussing a scenario with @ryangjchandler about his great alpine-tooltip (https://github.com/ryangjchandler/alpine-tooltip/issues/5).

I posed the question: Is there a way to make the text of the tooltip dynamic? I'm trying to figure out how to make it toggle between "Enable" and "Disable" based on an underlying Livewire property being true or false.

Something like
`x-data="{ tooltip: {{ $user->disabled ? "'Enable'" : "'Disable'" }} }"`

The tooltip is inside a Livewire component - a data table. It is correct when the page loads (e.g. "Disable"), and the DOM changes when the $user->disabled property is updated to true by Livewire (i.e. "Enable"), but the UI shows "Disable" anyway.

Could it be that Alpine isn't re-initializing the component when the DOM is patched by Livewire?

**Comment by SimoTod** [comment]
Alpine doesn't reinitialise the component, you need Livewire's `@entangle`

**Reply by ryangjchandler** [reply]
The old component should be destroyed and re-created though, as the `x-data` attribute is being changed, effectively creating a new component.

Another issue is with Livewire's pagination. If you have an `x-data` attribute inside of the paginated content, Alpine won't re-initialise / destroy the old component and create a new one when the page changes.

Alpine 2.x behaved in that way, not anymore it seems.

**Reply by SimoTod** [reply]
I'm not sure for v3, i think Livewire stops the mutation observer, updates the dom, clones the old component into the new one (so what's in the html becomes irrelevant) and then restarts the mutation observer.

**Reply by ryangjchandler** [reply]
Yeah, `@entangle` doesn't work too well with pagination etc so it's a bit annoying that it's not working the same way.

**Reply by SimoTod** [reply]
Mmmh, the pagination thing seems an undesired side effect of livewire/alpine trying to be too clever when patching the DOM.

**Comment by nijlandvpr** [comment]
And then there is the issue that this use case applies to rows in a table,
with each tooltip applying to an underlying discrete record.

I can't think of a way that @entangle can do that - I thought it references
public properties of the component, like $userId, not $user->id.

**Reply by nijlandvpr** [reply]
This works on each row that re-renders (also added wire:key="user-{{ $user->id }}" to each:

```
<div
    x-data="{ tooltip: {{ "'" . ($user->login_disabled ? 'Enable' : 'Disable') . "'" }} }"
    wire:click="disableUser({{ $user->id }})" class="cursor-pointer"
>
     @if ($user->login_disabled)
         @svg('bi-play-circle', ['class' => 'w-5 h-5 text-gray-500', 'x-tooltip' => 'tooltip' ])
     @else
         @svg('bi-stop-circle', ['class' => 'w-5 h-5 text-gray-500', 'x-tooltip' => 'tooltip' ])
      @endif
</div>

```

**Reply by ryangjchandler** [reply]
Yeah, I think this is probably going to be the simplest way to do it. FYI, you could use `json_encode()` too: `{{ json_encode($user->login_disabled ? 'Enable' : 'Disable') }}`

**Reply by nijlandvpr** [reply]
Even better - thanks

**Reply by nijlandvpr** [reply]
Unfortunately, I spoke too soon. Still does not refresh even though render runs. Needs a page reload.

**Reply by nijlandvpr** [reply]
With x-effect sending the tooltip to the console there is something I don't understand.

There are 6 rows of data.

Page reload send 6 logs, all 'Disable'.
Toggle the boolean and it logs 7 'Disable'
Toggle it a second time and it logs 'Enable' followed by 6 'Disable'

![Screenshot-20211007155215-480x181](https://user-images.githubusercontent.com/1731675/136460637-e4dfc61b-6368-4645-b92e-b4fde5c9d2bd.png)

Sigh - much to learn
